unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -1,43 +1,210 @@
+
+            model.wearables.Clear();
+
+            int wearablesCount = userProfile.avatar.wearables.Count;
+
+            if (!DataStore.i.common.isPlayerRendererLoaded.Get()) return;
+
+            for (var i = 0; i < wearablesCount; i++)
+            {
+                if (!wearablesCatalogService.WearablesCatalog.TryGetValue(userProfile.avatar.wearables[i], out var wearable))
+                {
+                    Debug.LogError($"Couldn't find wearable with ID {userProfile.avatar.wearables[i]}");
+                    continue;
+                }
+
+                model.wearables.Add(wearable);
+            }
+        }
+
+        private void UpdateAvatarPreview()
+        {
+            AvatarModel modelToUpdate = model.ToAvatarModel();
+
+            // We always keep the loaded emotes into the Avatar Preview
+            foreach (string emoteId in dataStore.emotesCustomization.currentLoadedEmotes.Get())
+                modelToUpdate.emotes.Add(new AvatarModel.AvatarEmoteEntry() { urn = emoteId });
+
+            view.UpdateAvatarPreview(modelToUpdate);
+        }
+
+        private void OnNewEmoteAdded(string emoteId) =>
+            UpdateAvatarPreview();
+
+        private void OnEmoteEquipped(string emoteId)
+        {
+            wearablesCatalogService.WearablesCatalog.TryGetValue(emoteId, out WearableItem equippedEmote);
+
+            if (equippedEmote != null && equippedEmote.IsEmote())
+                avatarIsDirty = true;
+        }
+
+        private void OnEmoteUnequipped(string emoteId)
+        {
+            wearablesCatalogService.WearablesCatalog.TryGetValue(emoteId, out WearableItem unequippedEmote);
+
+            if (unequippedEmote != null && unequippedEmote.IsEmote())
+                avatarIsDirty = true;
+        }
+
+        private void OnEmotePreviewed(string emoteId) =>
+            view.PlayPreviewEmote(emoteId);
+
+        private void EquipBodyShape(WearableItem bodyShape)
+        {
+            if (bodyShape.data.category != WearableLiterals.Categories.BODY_SHAPE)
+            {
+                Debug.LogError($"Item ({bodyShape.id} is not a body shape");
+                return;
+            }
+
+            if (model.bodyShape == bodyShape)
+                return;
+
+            model.bodyShape = bodyShape;
+            backpackEmotesSectionController.SetEquippedBodyShape(bodyShape.id);
+        }
+
+        private void EquipSkinColor(Color color) =>
+            model.skinColor = color;
+
+        private void EquipHairColor(Color color) =>
+            model.hairColor = color;
+
+        private void EquipEyesColor(Color color) =>
+            model.eyesColor = color;
+
+        private void SetRenderScale(bool visible)
+        {
+            // NOTE(Brian): SSAO doesn't work correctly with the offset avatar preview if the renderScale != 1.0
+            var asset = GraphicsSettings.renderPipelineAsset as UniversalRenderPipelineAsset;
```

Comment:
we should move this logic to the view since it relies to unity visual components

Reply:
I have moved the logic into the `BackpackPreviewPanel` view: https://github.com/decentraland/unity-renderer/pull/4933/commits/73afaa525ee72f8d21da8c4e61a91d4c93a6b800